securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/api/ConsentDetailsApiController.java

```diff
@@ -131,7 +134,18 @@ public ResponseEntity<ConsentDetails> getConsentDetails(String consentRequestJws
                 details.setClientName(apiClient.getName());
                 details.setUsername(consentClientRequest.getUser().getUserName());
                 details.setUserId(consentClientRequest.getUser().getId());
-                details.setAccounts(consentClientRequest.getAccounts());
+
+                details.setAccounts(accountService.getAccountsWithBalance(details.getUserId()));
+
+                // DebtorAccount is optional, but the PISP could provide the account identifier details for the PSU
+                // The accounts displayed in the RCS ui needs to be The debtor account if is provided in the consent otherwise the user accounts
+                // TODO debtorAccount logic for VRP sweeping payments case until it will fixed for all payments  https://github.com/secureapigateway/secureapigateway/issues/731
+                if (details.getIntentType().equals(IntentType.DOMESTIC_VRP_PAYMENT_CONSENT)) {
+                    if(Objects.nonNull(((PaymentsConsentDetails) details).getDebtorAccount())) {
+                        setDebtorAccountWithBalance((PaymentsConsentDetails) details, consentRequestJws, intentId);
+                    }
+                }
```

Comment:
Specific case for VRP sweeping payments where the debtorAccount is mandatory until we fix the debtorAccount as optional for all payments.